champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/assassin/sword/Sever.java

```diff
@@ -98,41 +98,46 @@ private void onInteract(Player player, LivingEntity ent) {
 
         int level = getLevel(player);
         if (level <= 0) {
-            return; // Skill not active
+            return;
         }
 
         final PlayerUseSkillEvent event = UtilServer.callEvent(new PlayerUseSkillEvent(player, this, level));
         if (event.isCancelled()) {
-            return; // Skill was cancelled
+            return;
         }
 
-        if(ent == null) return;
+        if (ent == null) {
+            player.getWorld().playSound(player.getLocation(), Sound.ENTITY_SPIDER_HURT, 1.0F, 1.5F);
+            UtilMessage.simpleMessage(player, getClassType().getName(), "You failed <green>%s", getName());
+            return;
+        }
 
-        if (UtilMath.offset(player, ent) <= 3.0) {
-            if (ent instanceof Player damagee) {
-                if (UtilPlayer.getRelation(player, damagee) == EntityProperty.FRIENDLY) {
-                    return;
-                }
-            }
+        boolean withinRange = UtilMath.offset(player, ent) <= hitDistance;
+        boolean isFriendly = false;
+        if (ent instanceof Player damagee) {
+            isFriendly = UtilPlayer.getRelation(player, damagee) == EntityProperty.FRIENDLY;
+        }
 
+        if (!withinRange || isFriendly) {
+            UtilMessage.simpleMessage(player, getClassType().getName(), "You failed <green>%s", getName());
+        } else {
             championsManager.getEffects().addEffect(ent, EffectType.BLEED, (long) getDuration(level) * 1000L);
             UtilMessage.simpleMessage(player, getClassType().getName(), "You severed <alt>" + ent.getName() + "</alt>.");
             UtilMessage.simpleMessage(ent, getClassType().getName(), "You have been severed by <alt>" + player.getName() + "</alt>.");
-        } else {
-            UtilMessage.simpleMessage(player, getClassType().getName(), "You failed <green>%s", getName());
         }
-        ent.getWorld().playSound(ent.getLocation(), Sound.ENTITY_SPIDER_HURT, 1.0F, 1.5F);
 
+        player.getWorld().playSound(player.getLocation(), Sound.ENTITY_SPIDER_HURT, 1.0F, 1.5F);
     }
 
     @Override
     public double getCooldown(int level) {
-        return cooldown;
+        return cooldown - cooldownDecreasePerLevel;
```

Comment:
not how its calculated

should be cooldown - ((level - 1) * cooldownDecreasePerLevel)